### DISCUSSION TITLE
Failed install M1/M2 extra steps

### DISCUSSION THREAD

**Discussion by zhywang17** [BradyAJohnston/MolecularNodes#200] [General]
Hi Brady, 
I was trying to follow the instructions for M1/M2 Mac extra steps, and when I try to run 
`./python3.10 -m pip install MDAnalysis --cache-dir ~/MDAnalysis-wheel
`this error came: 
```

gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -I/Users/brecht/dev/build_darwin/deps/Release/sqlite/include -I/Users/brecht/dev/build_darwin/deps/Release/bzip2/include -I/Users/brecht/dev/build_darwin/deps/Release/lzma/include -I/Users/brecht/dev/build_darwin/deps/Release/zlib/include -mmacosx-version-min=11.00 -arch arm64 -mmacosx-version-min=11.00 -arch arm64 -I/Applications/Blender.app/Contents/Resources/3.5/python/include/python3.10 -c Bio/Align/_aligners.c -o build/temp.macosx-11.00-arm64-cpython-310/Bio/Align/_aligners.o
      Bio/Align/_aligners.c:11:10: fatal error: 'Python.h' file not found
      #include "Python.h"
               ^~~~~~~~~~
      1 error generated.
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for biopython


```
Would you please help me to check this out? 
Thanks

**Comment by BradyAJohnston** [comment]
It seems that Blender is unable to find the pre-built `.whl` for MDAnalysis. 

Are you able to show the location and the contents of `~/MDAnalysis-wheel`? Might be that the previous steps failed and the binaries weren't compiled properly

**Comment by zhywang17** [comment]
Here it is :
```
(base) zhenyuwang@Zhenyu 3.5 % ls ~/MDAnalysis-wheel 
GridDataFormats-1.0.1-py3-none-any.whl			mrcfile-1.4.3-py2.py3-none-any.whl
MDAnalysis-2.2.0-cp310-cp310-macosx_10_9_x86_64.whl	msgpack-1.0.5-cp310-cp310-macosx_10_9_x86_64.whl
Pillow-9.5.0-cp310-cp310-macosx_10_10_x86_64.whl	networkx-3.1-py3-none-any.whl
biopython-1.81-cp310-cp310-macosx_10_9_x86_64.whl	numpy-1.24.3-cp310-cp310-macosx_10_9_x86_64.whl
contourpy-1.0.7-cp310-cp310-macosx_10_9_x86_64.whl	packaging-23.1-py3-none-any.whl
cycler-0.11.0-py3-none-any.whl				pyparsing-3.0.9-py3-none-any.whl
fasteners-0.18-py3-none-any.whl				python_dateutil-2.8.2-py2.py3-none-any.whl
fonttools-4.39.3-py3-none-any.whl			scipy-1.10.1-cp310-cp310-macosx_10_9_x86_64.whl
gsd-2.8.1-cp310-cp310-macosx_10_9_x86_64.whl		selfcheck
http							six-1.16.0-py2.py3-none-any.whl
joblib-1.2.0-py3-none-any.whl				threadpoolctl-3.1.0-py3-none-any.whl
kiwisolver-1.4.4-cp310-cp310-macosx_10_9_x86_64.whl	tqdm-4.65.0-py3-none-any.whl
matplotlib-3.7.1-cp310-cp310-macosx_10_12_x86_64.whl	wheels
mmtf_python-1.1.3-py2.py3-none-any.whl

```
Thanks for your kind reply!

**Reply by BradyAJohnston** [reply]
nothing seems obviously out of the ordinary. I would maybe suggest clearing out that file and trying the build & install again

**Reply by zhywang17** [reply]
I tried several time and all failed, I searched some case, it seems like gcc cannot locate python.h, maybe I should add the python to the gcclist somewhere, I need to find out. Anyway will keep trying, thanks!

**Reply by BradyAJohnston** [reply]
The error that you are getting is indeed an error with GCC and the python that ships with Blender. Blender is unable to install python packages into the python version that ships with Blender, as it doesn't have the required `python.h` and associated files that are required for compiling packages. 

To get around this, the installation instructions tell you to pre-compile the MDAnalysis package with a different version of python and then install the pre-compiled version into the version that ships with Blender. Without more detailed error messages, it seems like the pre-compiled version is falling to be installed, Blender is then attempting to compile it's own version which then also fails. I am unsure why. 

This thread (https://github.com/BradyAJohnston/MolecularNodes/issues/108) is discussing exactly this error. There are some other solutions which are provided, perhaps try some of those.

**Reply by zhywang17** [reply]
I really appreciate your effort to help around! Now I can use a Blender on Windows with MN installed to read the file and then send the project to Blender on Mac lol, works well for me, maybe I would just looking forward to your upcoming version that officially settle this issue

**Comment by YaoYinYing** [comment]
Hi @zhywang17 I guess you are currently using a rosetta2-mediated x86-64 conda and  python. To fix this ,you need to re-create a ARM based conda env for this wheel building.

Basicly, try this ` CONDA_SUBDIR=osx-arm64 conda create -n MN_PY310 python=3.10 -y` for the armed python env.

**Reply by zhywang17** [reply]
My environment is exactly like what you said but the solution is not work for me, maybe I should reinstall blender?

**Reply by YaoYinYing** [reply]
Yes you'd better do a completely clean reinstallation of blender and MN. Just make sure EVERYTHING is ARMed.

**Reply by YaoYinYing** [reply]
> the solution is not work for 

And if you can share some error logs this might be more clear for me to do this help
